homeassistant/components/switcher_kis/switch.py

```diff
@@ -0,0 +1,294 @@
+        """Return the datetime for the last update received by the device."""
+        ret = self._device_data.last_data_update  # type: datetime
+        return ret
+
+    @property
+    def last_state_change(self) -> datetime:
+        """Return the datetime for the last state change."""
+        ret = self._device_data.last_state_change  # type: datetime
+        return ret
+
+    @property
+    def device_name(self) -> str:
+        """Return the device's name."""
+        ret = self._device_data.name  # type: str
+        return ret
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True if entity has to be polled for state.
+
+        False if entity pushes its state to HA.
+        """
+        return False
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return "{}-{}".format(
+            self._device_data.device_id, self._device_data.mac_addr)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def state(self) -> str:
+        """Return the state of the entity."""
+        return (
+            STATE_ON if self._state == SWITCHER_STATE_ON
+            else STATE_OFF if self._state == SWITCHER_STATE_OFF
+            else STATE_UNKNOWN)
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        return self.state == STATE_ON
+
+    @property
+    def current_power_w(self) -> Optional[int]:
+        """Return the current power usage in W."""
+        ret = self._device_data.power_consumption  # type: Optional[int]
+        return ret
+
+    @property
+    def state_attributes(self) -> Dict:
+        """Return the optional state attributes."""
+        attribs = {}
+
+        for prop, attr in PROPERTIES_TO_ATTRIBUTES.items():
+            value = getattr(self, prop)
+            if value:
+                attribs[attr] = value
+
+        return attribs
+
+    @property
+    def icon(self) -> Optional[str]:
+        """Return the icon to use in the frontend, if any."""
+        return self._icon
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return not self.state == STATE_UNKNOWN
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return True if unable to access real state of the entity."""
+        return False
+
+    async def async_update_data(self, device_data: SwitcherV2Device) -> None:
+        """Update the entity data.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if self._self_initiated:
+            self._self_initiated = False
+        else:
+            self._device_data = device_data
+            self._state = self._device_data.state
+            await self._hass.async_create_task(self.async_update_ha_state())
+
+        return None
+
+    def turn_on(self, **kwargs: Dict) -> None:
```

Comment:
Remove this method.  This will not be called since you are implementing `async_turn_on`